src/cryptonote_core/service_node_list.cpp

```diff
@@ -278,15 +278,14 @@ namespace service_nodes
 
     // check the portions
 
-    uint64_t total = 0;
+    uint64_t portions_left = STAKING_PORTIONS;
     for (size_t i = 0; i < service_node_portions.size(); i++)
     {
-      if (service_node_portions[i] < MIN_PORTIONS)
+      uint64_t min_portions = std::min(portions_left, (uint64_t)MIN_PORTIONS);
```

Comment:
Do we sort the portions by descending order prior to this?

Reply:
This might be problematic if the user doesn't provide the "last" contributor last (the one which is allowed to contribute less than 25%).

Reply:
Make sure you sort them before generating the registration command.